systems/subscriber/registry/pkg/db/subscriber_repo_test.go

```diff
@@ -56,60 +57,399 @@ func (u UkamaDbMock) ExecuteInTransaction2(dbOperation func(tx *gorm.DB) *gorm.D
+		DOB:                   "07-03-2023",
 		ProofOfIdentification: "Driver's License",
 		IdSerial:              "ABC123",
 		Address:               "123 Main St.",
 		CreatedAt:             time.Now(),
 		UpdatedAt:             time.Now(),
 		DeletedAt:             nil,
 	}
+}
+
+func TestSubscriber_Add(t *testing.T) {
+	t.Run("SuccessWithoutNestedFunc", func(t *testing.T) {
+		// Arrange
+		db, mock, repo := setupTestDB(t)
+		defer db.Close()
+
+		subscriber := createTestSubscriber("John", "john@example.com")
+
+		mock.ExpectBegin()
+		mock.ExpectExec("INSERT INTO \"subscribers\"").WithArgs(
+			subscriber.SubscriberId,
+			subscriber.Name,
+			subscriber.NetworkId,
+			subscriber.Email,
+			subscriber.PhoneNumber,
+			subscriber.Gender,
+			subscriber.DOB,
+			subscriber.ProofOfIdentification,
+			subscriber.IdSerial,
+			subscriber.Address,
+			subscriber.CreatedAt,
+			subscriber.UpdatedAt,
+			subscriber.DeletedAt).WillReturnResult(sqlmock.NewResult(1, 1))
+		mock.ExpectCommit()
+
+		// Act
+		err := repo.Add(&subscriber, nil)
+
+		// Assert
+		assert.NoError(t, err)
+		assert.NoError(t, mock.ExpectationsWereMet())
+	})
+
+	t.Run("SuccessWithNestedFunc", func(t *testing.T) {
+		// Arrange
+		db, mock, err := sqlmock.New()
+		if err != nil {
+			t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
+		}
+		defer db.Close()
+		gdb, _ := gorm.Open(postgres.New(postgres.Config{
+			DSN:                  "sqlmock_db_0",
+			DriverName:           "postgres",
+			Conn:                 db,
+			PreferSimpleProtocol: true,
+		}), &gorm.Config{})
+		repo := int_db.NewSubscriberRepo(&UkamaDbMock{
+			GormDb: gdb,
+		})
 
-	mock.ExpectBegin()
-	mock.ExpectExec("INSERT INTO \"subscribers\"").WithArgs(
-		subscriber.SubscriberId,
-		subscriber.Name,
-		subscriber.NetworkId,
-		subscriber.Email,
-		subscriber.PhoneNumber,
-		subscriber.Gender,
-		subscriber.DOB,
-		subscriber.ProofOfIdentification,
-		subscriber.IdSerial,
-		subscriber.Address,
-		subscriber.CreatedAt,
-		subscriber.UpdatedAt,
-		subscriber.DeletedAt).WillReturnResult(sqlmock.NewResult(1, 1))
-	mock.ExpectCommit()
-
-	err = repo.Add(&subscriber, nil)
-	assert.NoError(t, err)
-	assert.NoError(t, mock.ExpectationsWereMet())
+		subscriber := int_db.Subscriber{
+			SubscriberId:          uuid.NewV4(),
+			Name:                  "Jane",
+			NetworkId:             uuid.NewV4(),
+			Email:                 "jane@example.com",
+			PhoneNumber:           "555-555-5556",
+			Gender:                "Female",
+			DOB:                   "15-06-1990",
+			ProofOfIdentification: "Passport",
+			IdSerial:              "XYZ789",
+			Address:               "456 Oak Ave.",
+			CreatedAt:             time.Now(),
+			UpdatedAt:             time.Now(),
+			DeletedAt:             nil,
+		}
+
+		nestedFuncCalled := false
+		nestedFunc := func(sub *int_db.Subscriber, tx *gorm.DB) error {
+			nestedFuncCalled = true
+			assert.Equal(t, subscriber.SubscriberId, sub.SubscriberId)
```

Comment:
Why do we have assertions inside a fake/mock nested function? Maybe I am missing something.